apps/sel4test-tests/src/arch/arm/tests/breakpoints.c

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2024, UNSW Sydney
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <autoconf.h>
+
+#ifdef CONFIG_HARDWARE_DEBUG_API
+
+#include <sel4/sel4.h>
+#include <utils/attribute.h>
+
+#include <arch/debug.h>
+#include "../../../helpers.h"
+#include "../../../test.h"
+#include "../../../tests/breakpoints.h"
+
+static int faulter_main(seL4_Word arg0, seL4_Word arg1, seL4_Word arg2, seL4_Word arg3)
+{
+    TEST_SINGLE_STEP_ASM();
+    /* NOTHING BELOW SHOULD EVER BE REACHED */
+    int *ptr = NULL;
+    *ptr = 0xBEEF;
+    return 0;
```

Comment:
Maybe say something like: "Should not be reached, but if it does, it wakes up the handler thread".